draft-rescorla-tls-esni.md

```diff
@@ -302,6 +310,11 @@ associated with `suite`.
 suite
 : The cipher suite used to encrypt the SNI.
 
+version
+: The version of ESNI that is in use. The value of 0x0000 represents this RFC
+version. The value of 0xFFnn represents previous Internet Draft versions,
+where nn is the revision of an earlier Internet Draft.
```

Comment:
I am not sure if we need to explicitly include the version of ESNIKeys in the TLS extension.

Current approach assumes that the server looks for the corresponding ESNIKeys by checking the value of `record_digests`. That will lead you to the ESNIKeys that contains a version. Having another version seems unnecessary complication to me.

Reply:
A server can support multiple versions of the draft; it is a list in the DNS record, and the record digest would be the same for all versions in that list.

You are correct that the record_digest would be updated when a server adds support for a new version of ESNI, but that doesn't indicate *which* version of ESNI the client is trying to use.